src/Proxy.NET/Routes/AzureAISearch.cs

```diff
@@ -0,0 +1,62 @@
+using System.Net;
+using Microsoft.AspNetCore.Mvc;
+using Proxy.NET.Models;
+using Proxy.NET.Services;
+
+namespace Proxy.NET.Routes;
+
+public static class AzureAISearch
+{
+    public static RouteGroupBuilder MapAzureAISearchRoutes(this RouteGroupBuilder builder)
+    {
+        builder.MapPost("/indexes/{index}/docs/search", ProcessRequestAsync).WithMetadata(new Auth(Auth.Type.ApiKey));
+        builder.MapPost("/indexes('{index}')/docs/search.post.search", ProcessRequestAsync).WithMetadata(new Auth(Auth.Type.ApiKey));
+        return builder;
+    }
+
+    private static async Task<IResult> ProcessRequestAsync(
+        [FromServices] ICatalogService catalogService,
+        [FromServices] IProxyService proxyService,
+        [FromServices] IRequestService requestService,
+        HttpContext context,
+        string index
+    )
+    {
+        var routePattern = (context.GetEndpoint() as RouteEndpoint)?.RoutePattern.RawText;
+        var extPath = routePattern?.Split("/indexes").Last();
+        var requestContext = requestService.GetRequestContext() as RequestContext;
+        var apiVersion = ApiVersion(context);
+
+        var requestString = await new StreamReader(context.Request.Body).ReadToEndAsync();
+        if (string.IsNullOrEmpty(requestString))
+            throw new HttpRequestException("Invalid JSON object in body of request", null, HttpStatusCode.BadRequest);
+
+        requestContext!.DeploymentName = index!;
+        var deployment = await catalogService.GetCatalogItemAsync(requestContext);
+
+        var url = GenerateEndpointUrl(deployment, extPath!, apiVersion);
+        var (responseContent, statusCode) = await proxyService.HttpPostAsync(url, deployment.EndpointKey, requestString, requestContext);
+        return new ProxyResult(responseContent, statusCode);
+    }
+
+    private static Uri GenerateEndpointUrl(Deployment deployment, string extPath, string apiVersion)
+    {
+        var baseUrl = $"{deployment.EndpointUrl.TrimEnd('/')}";
```

Comment:
```suggestion
        var baseUrl = deployment.EndpointUrl.TrimEnd('/');
```

String interpolation wasn't providing anything here.